lib/linked_list.rb

```diff
@@ -12,120 +12,273 @@ def initialize(value, next_node = nil)
+    
+    until current.nil?
+      return true if current.data == value
+      current = current.next
     end
-
-    # method to add a new node with the specific data value in the linked list
-    # insert the new node at the beginning of the linked list
-    def add_first(value)
-      raise NotImplementedError
+    
+    return false
+  end
+  
+  # method to return the max value in the linked list
+  # returns the data value and not the node
+  def find_max
+    current = @head
+    max = @head
+    
+    return nil if current.nil?
+    
+    until current.nil?
+      if current.data > max.data
+        max = current
+      end
+      current = current.next
     end
-
-    # method to find if the linked list contains a node with specified value
-    # returns true if found, false otherwise
-    def search(value)
-      raise NotImplementedError
+    
+    return max.data
+  end
+  
+  # method to return the min value in the linked list
+  # returns the data value and not the node
+  def find_min
+    current = @head
+    min = @head
+    
+    return nil if current.nil?
+    
+    until current.nil?
+      if current.data < min.data
+        min = current
+      end
+      current = current.next
     end
-
-    # method to return the max value in the linked list
-    # returns the data value and not the node
-    def find_max
-      raise NotImplementedError
+    
+    return min.data
+  end
+  
+  
+  # method that returns the length of the singly linked list
+  def length
+    current = @head
+    count = 0
+    
+    until current.nil?
+      count += 1
+      current = current.next
     end
-
-    # method to return the min value in the linked list
-    # returns the data value and not the node
-    def find_min
-      raise NotImplementedError
+    
+    return count
+  end
+  
+  # method that returns the value at a given index in the linked list
+  # index count starts at 0
+  # returns nil if there are fewer nodes in the linked list than the index value
+  def get_at_index(index)
+    current = @head
+    
+    index.times do |i|
+      return nil if current.nil?
+      current = current.next
     end
-
-
-    # method that returns the length of the singly linked list
-    def length
-      raise NotImplementedError
+    
+    return current.data
+  end
+  
+  # method to print all the values in the linked list
+  def visit
+    current = @head
+    
+    until current.nil?
+      print current.data
+      current = current.next
     end
-
-    # method that returns the value at a given index in the linked list
-    # index count starts at 0
-    # returns nil if there are fewer nodes in the linked list than the index value
-    def get_at_index(index)
-      raise NotImplementedError
+  end
+  
+  # method to delete the first node found with specified value
+  def delete(value)
+    current = @head
+    previous = nil
+    
+    return nil if current.nil?
+    
+    until current.data == value
+      return nil if current.nil?

```

Comment:
Is this line needed?  if `current` is `nil` the app would crash because the `until current.data` above would cause a problem.